### ISSUE TITLE
Client.DLL Impersonation Exploit

### ISSUE THREAD

**Issue by 1v7531** [Facepunch/garrysmod-issues#2365]
Upon decompiling Client.DLL an user can change the 'Steam ID' in their Client.DLL in which they can impersonate another player. Let us take Garry as an example. Changing the Steam ID to Garry's Steam ID lets people take advantage of plugins such as 'ULX' and other plugins for administration by players, if Garry has a Super-Admin rank, then the impersonator can use those privileges to falsely ban people and give the person being impersonated a bad reputation. On the other hand the person who is impersonating someone will get away just fine.

TL:DR please find a way to send the actual client's steam id in a different way.

**Comment by yupi2** [comment]
Doesn't the server verify that the player is authenticated with Steam servers so players can't do this?

**Comment by Kefta** [comment]
Nope

**Comment by willox** [comment]
Clients can only fool other clients in to believing they have somebody else's Steam ID. This means that permissions in addons such as ULX shouldn't be affected as server-side code will see their true ID. The exploit is fixed in the dev branch.
